lib/kafka_ex.ex

```diff
@@ -199,10 +199,15 @@ defmodule KafkaEx do
     offset      = Keyword.get(opts, :offset, 0)
     handler     = Keyword.get(opts, :handler, KafkaExHandler)
 
-    {:ok, pid}  = GenEvent.start_link
-    :ok         = GenEvent.add_handler(pid, handler, [])
-    send(worker_name, {:start_streaming, topic, partition, offset, pid, handler})
-    GenEvent.stream(pid)
+    stream      = GenServer.call(worker_name, {:create_stream, handler})
+    send(worker_name, {:start_streaming, topic, partition, offset, handler})
```

Comment:
I am thinking to make `start_streaming` a first class function, that creates a worker that starts with an `event_pid` in its state so when the worker crashes it can be properly restarted with an `event_pid` though this would make sense when I have completed the `consumer_group` work that I am doing so the worker can commit offset and be smart to know where to continue streaming from after a crash.